src/file_parsing_uq.jl

```diff
@@ -0,0 +1,502 @@
+               the parameters' prior distributions and constraints) as values
+`name` - parameter name
+
+Returns a `ParameterDistribution`
+"""
+function get_parameter_distribution(param_dict::Dict, name::AbstractString)
+
+    # Constructing a parameter distribution requires a prior distribution,
+    # a constraint, and a name.
+    prior = construct_prior(param_dict[name])
+    constraint = construct_constraint(param_dict[name])
+
+    return ParameterDistribution(prior, constraint, name)
+end
+
+"""
+get_parameter_distribution(param_dict, names)
+
+Constructs a `ParameterDistribution` for an array of parameters
+
+Args:
+`param_dict` - nested dictionary that has parameter names as keys and the
+               corresponding dictionary of parameter information (in particular,
+               the parameters' prior distributions and constraints) as values
+`names` - array of parameter names
+
+Returns a `ParameterDistribution` 
+"""
+function get_parameter_distribution(param_dict::Dict, names::AbstractVector{String})
+    param_dist_arr = map(names) do name
+        get_parameter_distribution(param_dict, name)
+    end
+    return combine_distributions(param_dist_arr)
+end
+
+
+"""
+construct_constraint(param_info)
+
+Extracts information on type and arguments of each constraint and uses that
+information to construct a `Constraint`.
+
+Args:
+`param_info` - dictionary with (at least) a key "constraint", whose value is
+               the parameter's constraint(s) (as parsed from TOML file)
+
+Returns a `Constraint`
+"""
+function construct_constraint(param_info::Dict)
+
+    @assert(haskey(param_info, "constraint"))
+    c = Meta.parse(param_info["constraint"])
+
+    if c.args[1] == Symbol("repeat")
+        # Constraints are given as a `repeat` expression defining a vector of
+        # constraints of the same kind
+        constr = collect_from_expr(c.args[2], "c", repeat=true)
+        n_constr = c.args[3] # number of repeated constraints
+        return repeat([constr], n_constr)
+
+    else 
+        return collect_from_expr(c, "c")
+    end
+end
+
+
+"""
+construct_prior(param_info)
+
+Extracts information on type and arguments of the prior distribution and use
+that information to construct an actual `Distribution`
+
+Args:
+`param_info` - dictionary with (at least) a key "prior", whose value is the
+               parameter's distribution(s) (as parsed from TOML file)
+
+Returns a distribution of type `Parameterized`, `Samples`, or
+`VectorOfParameterized`
+"""
+function construct_prior(param_info::Dict)
+
+    @assert(haskey(param_info, "prior"))
+    d = Meta.parse(param_info["prior"])
+
+    if d.args[1] == Symbol("VectorOfParameterized")
+        # There are multiple distributions described by a
+        # `VectorOfParameterized`
+        return get_vector_of_parameterized(d)
+
+    else
+        # Single distribution
+        return get_distribution_from_expr(d)
+    end
+end
```

Comment:
```suggestion
function construct_prior(param_info::Dict)
    @assert(haskey(param_info, "prior"))
    return eval(Meta.parse(param_info["prior"]))
end
```
Can we use this instead?

Reply:
Closing this after discussion with @charleskawczynski --- we'll leave the code as is for now and might change it later, once we have agreed on whether using `eval` is permissible.